addons/project/models/project_task_recurrence.py

```diff
@@ -79,11 +79,28 @@ def _get_recurrence_delta(self):
             f"{self.repeat_unit}s": self.repeat_interval
         })
 
+    def _copy_task_dependency(self, task_copy, task_original):
+        if 'task_mapping' not in self.env.context:
+            self = self.with_context(task_mapping=dict())
+
+        for i in range(len(task_original.child_ids)):
+            self._copy_task_dependency(task_copy.child_ids[i], task_original.child_ids[i])
+
+        task_mapping = self.env.context.get('task_mapping')
+        task_mapping[task_original.id] = task_copy.id
+        tasks_copied = task_mapping.values()
+        task_original.write({'depend_on_ids': [Command.unlink(t.id) for t in task_original.depend_on_ids if t.id in tasks_copied]})
+        task_original.write({'dependent_ids': [Command.unlink(t.id) for t in task_original.dependent_ids if t.id in tasks_copied]})
+        task_copy.write({'depend_on_ids': [Command.link(task_mapping.get(t.id, t.id)) for t in task_original.depend_on_ids]})
+        task_copy.write({'dependent_ids': [Command.link(task_mapping.get(t.id, t.id)) for t in task_original.dependent_ids]})
```

Comment:
linking and unlinking relation like that is not very efficient. It would be better to build a dict containing the necessary values and link it in one go.